amespahdbpythonsuite/amespahdb.py

```diff
@@ -0,0 +1,707 @@
+        **keywords
+            Arbitrary keyword arguments.
+
+        """
+
+        # Create MD5 hash function pickle.
+        md5 = tempfile.gettempdir() + '/' + hashlib.md5(open(filename, 'r').read().encode()).hexdigest()
+
+        joined = md5 + '_joined.pkl'
+
+        if not keywords.get('cache', True):
+            if os.path.isfile(joined) and os.access(joined, os.R_OK):
+                os.remove(joined)
+
+        md5 += '.pkl'
+
+        # Check if database is dumped in cache and restore it.
+        if keywords.get('cache', True) and os.path.isfile(md5) and os.access(md5, os.R_OK):
+
+            self.message('RESTORING DATABASE FROM CACHE')
+
+            # start timer
+            tstart = time.perf_counter()
+
+            # Open and read the dumped database.
+            with open(md5, 'rb') as f:
+                self.__data = pickle.load(f)
+
+            # Store the dumped database filename.
+            self.__data['filename'] = md5
+
+            # stop timer and calculate elapsed time
+            elapsed = timedelta(seconds=(time.perf_counter() - tstart))
+
+            info = [f'FILENAME                    : {md5}',
+                    f'ORIGNINAL FILENAME          : {filename}',
+                    f'PARSE TIME                  : {elapsed}',
+                    f'VERSION (DATE)              : {self.__data["version"]} ({ self.__data["date"]})',
+                    f'COMMENT                     : {self.__data["comment"]}']
+            self.message(info, space=0)
+
+        else:
+
+            self.message('PARSING DATABASE: THIS MAY TAKE A FEW MINUTES')
+
+            # start timer
+            tstart = time.perf_counter()
+
+            # Call XMLparser module to parse the database.
+            parser = XMLparser(filename=filename, validate=keywords.get('check', True))
+
+            # Store the database into self.__self.data.
+            self.__data = parser.to_pahdb_dict()
+
+            # Dump the database into pickle in the cache directory.
+            with open(md5, 'wb') as f:
+                pickle.dump(self.__data, f, pickle.HIGHEST_PROTOCOL)
+
+            # Store the dumped database filename.
+            self.__data['filename'] = md5
+
+            # stop timer and calculate elapsed time
+            elapsed = timedelta(seconds=(time.perf_counter() - tstart))
+            print(f'Elapsed time: {elapsed}\n')
+
+    def pointer(self):
+        return self.__data
+
+    def __getkeybyuids(self, key, uids):
+        """
+        Get a dictionary of PAHdb properties retrieved by keyword for provided UIDs.
+
+        Parameters
+        ----------
+        key : str
+            Database keyword.
+        uids : list of integers
+            List of UIDs.
+
+        Returns
+        -------
+        Dictionary of retrieved properties with UIDs as keys.
+
+        """
+        return copy.deepcopy(dict((uid, self.__data['species'][uid][key]) for uid in uids))
+
+    def subdatabase(self, uids):
```

Comment:
This method's use  is commented out. If it is not needed, it can be removed.

Reply:
Should we keep it for potential future use, or completely remove it?

Reply:
I'm not really seeing what it be used for; if a subset of data is wanted it can be obtained through other means. I suggest to remove it for now and reintroduce it if it turns out we do really need it.